test/ERC20VotesPartialDelegationUpgradeable.t.sol

```diff
@@ -76,30 +78,98 @@ contract PartialDelegationTest is Test {
     return delegations;
   }
 
-  /// @dev
-  function _createValidPartialDelegation(uint256 _n, uint256 _seed) internal view returns (PartialDelegation[] memory) {
-    _seed = bound(
-      _seed,
-      1,
-      /* private key can't be bigger than secp256k1 curve order */
-      115_792_089_237_316_195_423_570_985_008_687_907_852_837_564_279_074_904_382_605_163_141_518_161_494_337 - 1
-    );
-    _n = _n != 0 ? _n : (_seed % tokenProxy.MAX_PARTIAL_DELEGATIONS()) + 1;
-    PartialDelegation[] memory delegations = new PartialDelegation[](_n);
-    uint96 _totalNumerator;
-    for (uint256 i = 0; i < _n; i++) {
-      uint96 _numerator = uint96(
-        bound(
-          uint256(keccak256(abi.encode(_seed + i))) % tokenProxy.DENOMINATOR(), // initial value of the numerator
-          1,
-          tokenProxy.DENOMINATOR() - _totalNumerator - (_n - i) // ensure that there is enough numerator left for the
-            // remaining delegations
-        )
-      );
-      delegations[i] = PartialDelegation(address(uint160(uint160(vm.addr(_seed)) + i)), _numerator);
-      _totalNumerator += _numerator;
+  function _expectEmitDelegateChangedEvents(
```

Comment:
This helper looks like it recreates the logic that exists in the contract which would make any test using this helper very brittle to regressions and also increases the likelihood that a bug in the implementation will be reflected in the underlying test. If it is possible it would probably be better to simplify the the tests.

Reply:
thanks @alexkeating , agreed -- @garyghayrat and I just wrote some concrete unit tests for this helper, in order to strengthen our confidence here. the event emission *is* fairly nuanced, so i think we'll need a helper here that we're confident in.